iothub/device/samples/PnpDeviceSamples/Thermostat/Program.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Shared;
+using PnpHelpers;
+
+namespace TemperatureController
+{
+    public class Program
+    {
+        private const string DeviceConnectionString = "device_connection_string_here";
+        private const string ModelId = "dtmi:com:example:TemperatureController;1";
```

Comment:
why not readonly?

Reply:
you mean, define this as a `static readonly`, as opposed to a `const`? A `const` is always `static`, by default., and is preferred when you want to assign a constant value (available at compile time) to a variable. 

One place where I find with static readonly fields useful, is when you want to initialize it only when the class instance is accessed for the first time (if it is a resource heavy operation, etc).

You can reference this stackoverflow answer for more details: https://stackoverflow.com/questions/55984/what-is-the-difference-between-const-and-readonly-in-c